src/fiber/channel.rs

```diff
@@ -1304,9 +1304,18 @@ where
     ) -> ProcessingChannelResult {
         state.check_for_tlc_update(None, true, false)?;
         state.check_remove_tlc_with_reason(TLCId::Received(command.id), &command.reason)?;
-        state
+        let payment_hash = state
             .tlc_state
             .set_received_tlc_removed(command.id, command.reason.clone());
+        if let RemoveTlcReason::RemoveTlcFulfill(RemoveTlcFulfill { payment_preimage }) =
+            command.reason
+        {
+            self.store
```

Comment:
There is the same store logic in `handle_remove_tlc_peer`, maybe it's should be moved to `apply_remove_tlc_operation`, which will invoked after two peers singed, it's both for inbound and outbound tlcs.

Reply:
let me create a new PR for refactoring, thanks.